consensus/istanbul/core/preprepare.go

```diff
@@ -86,13 +86,14 @@ func (c *core) handlePreprepare(msg *istanbul.Message) error {
 	if err := c.checkMessage(istanbul.MsgPreprepare, preprepare.View); err != nil {
 		if err == errOldMessage {
 			// Get validator set for the given proposal
-			valSet := c.backend.ParentBlockValidators(preprepare.Proposal).Copy()
-			previousProposer := c.backend.AuthorForBlock(preprepare.Proposal.Number().Uint64() - 1)
-			valSet.CalcProposer(previousProposer, preprepare.View.Round.Uint64())
+			valSet := c.backend.ParentBlockValidators(preprepare.Proposal)
```

Comment:
Why remove the `Copy`?

Reply:
Copy was there because `CalcProposer` mutates the valSet. Now a valSet is immutable, so we don't need to copy it